### PR TITLE
feat(account-lib): add claim for dot staking

### PR DESCRIPTION
**Pull request by yangran0**
Ticket: STLX-14695

### CODE REVIEW

File: modules/account-lib/src/coin/dot/claimBuilder.ts [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+      {
+        validatorStash: this._validatorStash,
+        era: this._claimEra,
+      },
+      baseTxInfo.baseTxInfo,
+      baseTxInfo.options,
+    );
+  }
+
+  protected get transactionType(): TransactionType {
```
Comment (byhow) [inline modules/account-lib/src/coin/dot/claimBuilder.ts:43]: nit: missing some jsdocs for some of the functions

File: modules/account-lib/src/coin/dot/claimBuilder.ts [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    return tx;
+  }
+
+  /** @inheritdoc */
+  validateTransaction(_: Transaction): void {
+    super.validateTransaction(_);
+    this.validateFields(this._claimEra, this._validatorStash);
+  }
+
+  private validateFields(claimEra: string, validatorStash: string): void {
```
Comment (byhow) [inline modules/account-lib/src/coin/dot/claimBuilder.ts:106]: ditto

File: modules/account-lib/src/coin/dot/claimBuilder.ts (hunk 1 above)
Comment (yangran0) [reply modules/account-lib/src/coin/dot/claimBuilder.ts:46]: done

File: modules/account-lib/src/coin/dot/claimBuilder.ts (hunk 2 above)
Comment (yangran0) [reply modules/account-lib/src/coin/dot/claimBuilder.ts:118]: done

**Review by byhow — APPROVED** [review]
LGTM

**Pull request merged**